test/providers/provider_level_zero.cpp

```diff
@@ -553,6 +556,88 @@ TEST_P(umfLevelZeroProviderTest, setDeviceOrdinalValid) {
+    ASSERT_NE(provider, nullptr);
+
+    umf_memory_pool_handle_t pool = NULL;
+    umf_result = umfPoolCreate(umfProxyPoolOps(), provider, NULL, 0, &pool);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+
+    size_t size = 1024;
+    void *ptr = umfPoolMalloc(pool, size);
+    ASSERT_NE(ptr, nullptr);
+
+    umf_memory_properties_handle_t props_handle = NULL;
+    umf_result_t result = umfGetMemoryPropertiesHandle(ptr, &props_handle);
+    ASSERT_EQ(result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(props_handle, nullptr);
+
+    umf_usm_memory_type_t type = UMF_MEMORY_TYPE_UNKNOWN;
+    result = umfGetMemoryProperty(
+        props_handle, UMF_MEMORY_PROPERTY_POINTER_TYPE, sizeof(type), &type);
+    ASSERT_EQ(result, UMF_RESULT_SUCCESS);
+    ASSERT_EQ(type, umfExpectedMemoryType);
+
+    // base address and size
+    void *baseAddress = nullptr;
+    result =
+        umfGetMemoryProperty(props_handle, UMF_MEMORY_PROPERTY_BASE_ADDRESS,
+                             sizeof(baseAddress), &baseAddress);
+    ASSERT_EQ(result, UMF_RESULT_SUCCESS);
+    ASSERT_EQ(baseAddress, ptr);
+
+    size_t baseSize = 0;
+    result = umfGetMemoryProperty(props_handle, UMF_MEMORY_PROPERTY_BASE_SIZE,
+                                  sizeof(baseSize), &baseSize);
+    ASSERT_EQ(result, UMF_RESULT_SUCCESS);
+    ASSERT_GE(baseSize, size);
+
+    int64_t bufferId = 0;
+    result = umfGetMemoryProperty(props_handle, UMF_MEMORY_PROPERTY_BUFFER_ID,
+                                  sizeof(bufferId), &bufferId);
+    ASSERT_EQ(result, UMF_RESULT_SUCCESS);
+    ASSERT_GE(bufferId, 0);
+
+    if (umfExpectedMemoryType != UMF_MEMORY_TYPE_HOST) {
+        ze_device_handle_t device = nullptr;
+        result = umfGetMemoryProperty(props_handle, UMF_MEMORY_PROPERTY_DEVICE,
+                                      sizeof(device), &device);
+        ASSERT_EQ(result, UMF_RESULT_SUCCESS);
+        ASSERT_EQ(device, l0TestHelper.get_test_device());
+    }
+
+    ze_context_handle_t context = nullptr;
+    result = umfGetMemoryProperty(props_handle, UMF_MEMORY_PROPERTY_CONTEXT,
+                                  sizeof(context), &context);
+    ASSERT_EQ(result, UMF_RESULT_SUCCESS);
+    ASSERT_EQ(context, l0TestHelper.get_test_context());
+
+    // check the props of pointer from the middle of alloc
+    void *midPtr = static_cast<char *>(ptr) + size / 2;
+    result = umfGetMemoryPropertiesHandle(midPtr, &props_handle);
+    ASSERT_EQ(result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(props_handle, nullptr);
+    result = umfGetMemoryProperty(
+        props_handle, UMF_MEMORY_PROPERTY_POINTER_TYPE, sizeof(type), &type);
+    ASSERT_EQ(result, UMF_RESULT_SUCCESS);
+    ASSERT_EQ(type, umfExpectedMemoryType);
+
+    result =
+        umfGetMemoryProperty(props_handle, UMF_MEMORY_PROPERTY_BASE_ADDRESS,
+                             sizeof(baseAddress), &baseAddress);
+    ASSERT_EQ(result, UMF_RESULT_SUCCESS);
+    ASSERT_EQ(baseAddress, ptr);
+
```

Comment:
I'm missing an invalid test for `umfGetMemoryProperty` with wrong mem_property_id

Reply:
TEST_P(ProviderPropsTest, invalidPropertyId)